### PR TITLE
Add support for addons

### PR DESCRIPTION
**Pull request by sabrichu**
Depends on https://github.com/usdigitalresponse/storefront/pull/37

Adds an `Add-On Item` badge to items that have the `Add On` property. Users are only allowed to add these items if they have at least one regular item in their cart. Removing the last regular item in their cart removes all add-on items.

Tested with and without the `Add On` column

**Airtable schema changes**

- new field in `Inventory` table: `Add On`, type `Checkbox`

Screen Shot 2020-08-08 at 8 07 40 PM
Screen Shot 2020-08-08 at 8 07 21 PM
Screen Shot 2020-08-08 at 8 07 10 PM

### CODE REVIEW

File: src/api/get-cms.js [hunk 1]
```diff
@@ -40,6 +40,7 @@ exports.handler = async (event, context) => {
           }, {}),
           price: row.fields['Price'],
           image: row.fields['Image'],
+          addOn: row.fields['Add On'] != null ? row.fields['Add On'] : false,
```
Comment (trip41) [inline src/api/get-cms.js:43]: nit: `!!row.fields['Add On']` just as good here

File: src/common/types.ts [hunk 2]
```diff
@@ -43,6 +43,7 @@ export interface InventoryRecord {
   strings?: Record<string, InventoryStrings>;
   price: number;
   image: AirtableImage[];
+  addOn?: boolean;
```
Comment (trip41) [inline src/common/types.ts:46]: with what you added in get-cms.js this isn't optional, this is always present, so can remove the `?`

File: src/common/types.ts [hunk 3]
```diff
@@ -83,6 +84,7 @@ export interface INavItem {
 export interface IOrderItem {
   id: string;
   quantity: number;
+  addOn?: boolean;
```
Comment (trip41) [inline src/common/types.ts:87]: same as above, can remove `?`

File: src/components/ProductDetail.tsx [hunk 4]
```diff
@@ -97,9 +100,26 @@ const ProductDetail: React.FC<Props> = ({ card, product, className }) => {
...
-                <Content id="products_purchase_button_label" />
-              </Button>
+              <Tooltip
+                title={
+                  addOn && orderItemCount === 0
+                    ? 'This is an add-on item. Add another item to your cart to include this item'
+                    : ''
+                }
+              >
+                <span>
```
Comment (trip41) [inline src/components/ProductDetail.tsx:110]: why do we need an extra `<span>` in here?

Comment (sabrichu) [reply src/components/ProductDetail.tsx:110]: Ah I can add a comment. `Tooltip` doesn't show up if its immediate child is `disabled`, so needed a buffer element

File: src/common/types.ts (hunk 3 above)
Comment (sabrichu) [reply src/common/types.ts:87]: Oh actually in this case `addOn` is that's not always there, it's just used in the `AddItem` and `RemoveItem` steps but not everywhere we use `IOrderItem`

Comment (trip41) [reply src/common/types.ts:87]: ah got it cool

**Pull request merged**